tests/server/routers/test_utils.py

```diff
@@ -6,6 +6,22 @@
 
 import pytest
 
+import re
```

Comment:
I'm still not entirely sure what is causing this bug to trigger in these dependency updates. Could you perhaps make a separate PR that just includes this test, so we can see if it fails? I'd also recommend just sticking it at the end of `tests.servers.routers.test_structures` rather than in the `test_utils.py` file.

Reply:
The bug is triggered by pydantic 1.10.  Basically, pydantic now also encodes the query part of the url. 
That was however already done in our code with urllib.parse.encode.  So we have to turn of one of the two.

See more info at https://github.com/pydantic/pydantic/issues/3061and https://github.com/pydantic/pydantic/pull/4224
And this bug report: https://github.com/pydantic/pydantic/issues/4458
It looks like they are already working on a fix : https://github.com/pydantic/pydantic/pull/4461
Although there is still some discussion about whether it should be implemented. 

There are also still some problems with spaces. Not all browsers can handle a + sign correctly. So my solution may not work for everybody. I also tried to keep the spaces, but then I get an error:

status | "500"
-- | --
code | "value_error.url.extra"
title | "ValidationError"
detail | "URL invalid, extra characters found after valid URL: ' HAS \"Ac\"&page_limit=2&page_offset=2'"
source |  
pointer | "/next"

So maybe we should wait to see if it gets accepted.


This test mostly checks whether the next link is in a format that will be handled correctly by pydantic 1.10.
It does not test whether the next link is valid. For that, we should use `_test_page_limit` from validator.py

I deliberately put it in test_utils.py as it applies to all endpoints and not just structures and the affected code is also in utils.py. I can use the EntryResponseMany class instead of the StructureResponseMany class if you prefer.

Reply:
Yeah, let's wait. It looks like the main author is working on a fix (as of 30 mins ago).

Reply:
I still think it would be useful to add the test to `test_structures.py` in a separate PR though.